src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/TangibleAssetsNoteServiceImpl.java

```diff
@@ -0,0 +1,126 @@
+
+    private static final String INVALID_URI_MESSAGE = "Invalid URI for tangible assets note resource";
+
+    @Override
+    public TangibleAssets getTangibleAssets(String transactionId, String companyAccountsId,
+        String companyNumber) throws ServiceException {
+
+        ApiClient apiClient = apiClientService.getApiClient();
+
+        String uri = TANGIBLE_ASSET_NOTE.expand(transactionId, companyAccountsId).toString();
+
+        TangibleApi tangibleApi;
+
+        try {
+            tangibleApi = apiClient.smallFull().tangible().get(uri).execute();
+        } catch (ApiErrorResponseException e) {
+
+            if (e.getStatusCode() == HttpStatus.NOT_FOUND.value()) {
+                return new TangibleAssets();
+            }
+
+            throw new ServiceException("Error retrieving tangible assets note resource", e);
+        } catch (URIValidationException e) {
+
+            throw new ServiceException(INVALID_URI_MESSAGE, e);
+        }
+
+        return tangibleAssetsTransformer.getTangibleAssets(tangibleApi);
+
+    }
+
+    @Override
+    public List<ValidationError> postTangibleAssets(String transactionId, String companyAccountsId,
+        TangibleAssets tangibleAssets, String companyNumber) throws ServiceException {
+
+        ApiClient apiClient = apiClientService.getApiClient();
+
+        String uri = TANGIBLE_ASSET_NOTE.expand(transactionId, companyAccountsId).toString();
+
+        String smallFullUri = SMALL_FULL_URI.expand(transactionId, companyAccountsId).toString();
+        SmallFullApi smallFullApi = getSmallFullData(apiClient, smallFullUri);
+
+        TangibleApi tangibleApi = tangibleAssetsTransformer.getTangibleApi(tangibleAssets);
+
+        boolean tangibleResourceExists = hasTangibleAssetNote(smallFullApi.getLinks());
+
+        try {
+            if (!tangibleResourceExists) {
+                apiClient.smallFull().tangible().create(uri, tangibleApi).execute();
+            } else {
+                apiClient.smallFull().tangible().update(uri, tangibleApi).execute();
+            }
+        } catch (URIValidationException e) {
+            throw new ServiceException(INVALID_URI_MESSAGE, e);
+        } catch (ApiErrorResponseException e) {
+            if (e.getStatusCode() == HttpStatus.BAD_REQUEST.value()) {
+                List<ValidationError> validationErrors = validationContext.getValidationErrors(e);
+                if (validationErrors.isEmpty()) {
+                    throw new ServiceException(
+                        "Bad request when creating tangible assets note resource", e);
+                }
+                return validationErrors;
+            }
+            throw new ServiceException("Error creating tangible assets note resource", e);
+        }
+
+        return new ArrayList<>();
+    }
+
+    private SmallFullApi getSmallFullData(ApiClient apiClient, String smallFullUri)
```

Comment:
This logic already exists in the `SmallFullService`

Reply:
A method with this signature does not exist in the `SmallFullService`.

Reply:
You can use `getSmallFullAccounts` in the `SmallFullService`  - this means you can remove the `smallFullUri` also.

Reply:
Resolved as it is inefficient to use the current `SmallFullService` method as it results in multiple ApiClient objects being created. This will likely be refactored in the future as it is not in the scope of the story.